Source/SourceKittenFramework/File.swift

```diff
@@ -0,0 +1,263 @@
+        if !shouldParseDeclaration(dictionary) {
+            return nil
+        }
+        let offset = Int(SwiftDocKey.getOffset(dictionary)!)
+        let lineBreakIndexBeforeOffset = indexBeforeValue(offset, inArray: lineBreaks)
+        let previousLineBreakOffset = lineBreaks[lineBreakIndexBeforeOffset] + 1
+        if let bodyOffset = SwiftDocKey.getBodyOffset(dictionary) {
+            return contents.filteredSubstring(previousLineBreakOffset, end: Int(bodyOffset))
+        }
+        let nextLineBreakOffset = (lineBreakIndexBeforeOffset + 1 < lineBreaks.count) ? lineBreaks[lineBreakIndexBeforeOffset + 1] : lineBreaks.last!
+        return contents.filteredSubstring(previousLineBreakOffset, end: nextLineBreakOffset + 1)
+    }
+
+    public func markNameFromDictionary(dictionary: XPCDictionary) -> String? {
+        precondition(SwiftDocKey.getKind(dictionary)! == SyntaxKind.CommentMark.rawValue)
+        let offset = Int(SwiftDocKey.getOffset(dictionary)!)
+        let length = Int(SwiftDocKey.getLength(dictionary)!)
+        if let fileContentsData = contents.dataUsingEncoding(NSUTF8StringEncoding) {
+            let subdata = fileContentsData.subdataWithRange(NSRange(location: offset, length: length))
+            if let substring = NSString(data: subdata, encoding: NSUTF8StringEncoding) as String? {
+                return substring
+            }
+        }
+        return nil
+    }
+
+    /**
+    Process a SourceKit editor.open response dictionary by removing undocumented tokens with no
+    documented children. Add cursor.info information for declarations. Add name to mark comments.
+
+    :param: dictionary        `XPCDictionary` to mutate.
+    :param: cursorInfoRequest SourceKit xpc dictionary to use to send cursorinfo request.
+
+    :returns: Whether or not the dictionary should be kept.
+    */
+    public func processDictionary(var dictionary: XPCDictionary, cursorInfoRequest: xpc_object_t? = nil) -> XPCDictionary {
+        // Update substructure
+        if let substructure = newSubstructure(dictionary, cursorInfoRequest: cursorInfoRequest) {
+            dictionary[SwiftDocKey.Substructure.rawValue] = substructure
+        }
+
+        if let cursorInfoRequest = cursorInfoRequest {
+            if let updateDict = updateDict(dictionary, cursorInfoRequest: cursorInfoRequest) {
+                for (key, value) in updateDict {
+                    dictionary[key] = value
+                }
+            }
+        }
+
+        if let parsedDeclaration = parseDeclaration(dictionary) {
+            dictionary[SwiftDocKey.ParsedDeclaration.rawValue] = parsedDeclaration
+        }
+        return dictionary
+    }
+
+    public func furtherProcessDictionary(var dictionary: XPCDictionary, documentedTokenOffsets: [Int], cursorInfoRequest: xpc_object_t) -> XPCDictionary {
+        let offsetMap = generateOffsetMap(documentedTokenOffsets, dictionary: dictionary)
+        for offset in offsetMap.keys.array.reverse() {
```

Comment:
I honestly don't remember. I should have written a comment about that decision when wrote the code, obviously. I'll have to investigate at some point.